src/lib.rs

```diff
@@ -492,36 +556,50 @@ pub mod circuit_graph {
             // path. (KVL)
             for (i, path) in paths.iter().enumerate() {
                 // Write -1 for the coefficient of the source voltage
-                coeffs.write(
-                    i,
-                    num_unknown_currents + path.0.index(),
-                    T::faer_one().faer_neg(),
-                );
+                unsafe {
+                    coeffs.write_unchecked(
+                        i,
+                        num_unknown_currents + path.0.index(),
+                        T::faer_one().faer_neg(),
+                    );
+                };
```

Comment:
Same for all the `write` calls, but why have they been changed like this?

My assumption is that we've guaranteed we're not going to try to write outside the bounds of the matrix's indices, so we can get away with taking a minor performance advantage from not checking those bounds?

Reply:
Correct; advice from Sarah on this is to basically always use the unchecked versions of rea/write ops if safety is already guaranteed. It would seem that it can add up quite a bit if the checks are inside loops.